eks/s3.tf

```diff
@@ -1,7 +1,62 @@
+data "aws_caller_identity" "current" {}
+
+resource "aws_kms_key" "data_bucket_key" {
+  description             = "This key is used to encrypt the data_bucket"
+  deletion_window_in_days = 10
+
+
+  # This policy allows the root to manage the keys - needed for terraform to manage the keys and provides the cluster nodes with limited access to keys
+  policy = <<EOF
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Action": [
+        "kms:*"
+      ],
+      "Principal": { "AWS": [
+          "arn:aws:iam::${data.aws_caller_identity.current.account_id}:root"
+        ]
+      },
+      "Effect": "Allow",
+      "Resource": "*"
+    },
+    {
+      "Action": [
+        "kms:Decrypt",
+        "kms:GenerateDataKey*"
+      ],
+      "Principal": { "AWS": [
+          "${module.eks-cluster.worker_iam_role_arn}"
+        ]
+      },
+      "Effect": "Allow",
+      "Resource": "*"
```

Comment:
This is fairly heavy handed. Is it possible to narrow down what we're allowed to decrypt? Would be preferable to make the scope s3 bucket specific.

Reply:
switching to default encryption